### PR TITLE
Copy example config files on server startup if required

### LINKED ISSUE #48: Server config
The server configuration is currently a mild irritant.
See also [this comment](https://github.com/openmsupply/open-msupply/pull/39#discussion_r880317456)

Every time I change branches, I have to reapply my config changes ( I'm not running the config which is currently checked in ) and on the demo server, I run a patch to correct the config after fetching latest.
( It also seems strange having configuration committed to the repo )

This is annoying.. I wondered about:

- rename base.yaml to base.yaml.example
- rename local.yaml to local.yaml.example
- on server startup, if the files don't exist, copy the examples over

### PR DESCRIPTION
**Pull request by jmbrunskill**
Closes #48

### CODE REVIEW

**Comment by jmbrunskill** [comment]
Would it make sense to only do this for the local.yaml file, and leave base.yaml as it was? 
Then we would move settings like sync username password into local/production.yaml?

**Comment by mark-prins** [comment]
sweet! I can feel the daily frustration just melting away.. 
+1 for Clemen's suggestion - I'd also add that the example should be expanded to include all possible options, but commented out - it's more helpful as a starting point, no?

e.g.

```
server:
  port: 8000
  # debug_no_access_control: true # enable this if you want to ignore API authorisation (dummy user will be used for all operations)
  # danger_allow_http: true # allow http in production mode
  # cors_origins: [http://localhost:8000] # Comma separated list of URLs; Used to set the allowed Origin in Cross Origin Request Security
sync:
  url: "http://localhost:2048"
  # username: "xxxxxxx"
  # password_sha256: "xxxxxxxxxxxxx"
  # interval_sec: 300
  # central_server_site_id: 1
  # site_id: 2
  # site_hardware_id: ""
# database:
  # host: "localhost"
  # port: 5432
  # username: "xxxx"
  # password: "xxxx"
  # database_name: "omsupply.sqlite"
```

**Comment by jmbrunskill** [comment]
Updated based on suggestions.
@andreievg did mention at lunch he has plans to try to get rid of everything but DB config in the yaml files.
Potentially even db config that could be handled with a .env file and DATABASE_URL environment variable (This would be compatible with diesel CLI as well)

I suggest we merge this in the meantime 
BTW: this may need some merging with PR #39 as there's some changes to config.yaml s there

File: server/configuration/base.yaml [hunk 1]
```diff
@@ -2,7 +2,11 @@ server:
   port: 8000
   # debug_no_access_control: true # enable this if you want to ignore API authorisation (dummy user will be used for all operations)
   # debug_cors_permissive: true # enable this to allow any origin for development purposes
-  cors_origins: [http://localhost:3003, https://demo-open.msupply.org, http://localhost:8000] # Used to set the allowed Origin in Cross Origin Requst Security
+  cors_origins: [
+      http://localhost:3003,
+      https://demo-open.msupply.org,
+      http://localhost:8000,
+    ] # Used to set the allowed Origin in Cross Origin Requst Security
 sync:
```
Comment (mark-prins) [inline server/configuration/base.yaml:10]: I'm using the sync config from the database.. so I'll still have to update `base.yaml` on every checkout in order to comment out the sync settings. I know that I said I agreed with Clemens and that we should still have base.yaml etc.. but I'm realising the implications now. 

Retaining base.yaml in the repo will still be irritating for me..

**Review by mark-prins — APPROVED** [review]
👍  LGTM - thanks James!

File: server/configuration/base.yaml (hunk 1 above)
Comment (jmbrunskill) [reply server/configuration/base.yaml:10]: Do we need sync settings in base.yaml? We could just remove them and people can add them if required in local.yaml?

Comment (jmbrunskill) [reply server/configuration/base.yaml]: Removed sync from base.yaml

**Pull request merged**